internal/testrunner/runners/system/runner.go

```diff
@@ -650,16 +650,23 @@ func writeSampleEvent(path string, doc common.MapStr) error {
 
 func validateFields(docs []common.MapStr, fieldsValidator *fields.Validator, dataStream string) error {
 	var multiErr multierror.Error
-	for _, doc := range docs {
-		if message, err := doc.GetValue("error.message"); err != common.ErrKeyNotFound {
-			multiErr = append(multiErr, fmt.Errorf("found error.message in event: %v", message))
-			continue
-		}
 
-		errs := fieldsValidator.ValidateDocumentMap(doc)
-		if errs != nil {
-			multiErr = append(multiErr, errs...)
-			continue
+	// prevents panic and performs cleanup/teardown for cases where
+	// no hits were found for a data stream
+	if len(docs) == 0 {
+		multiErr = append(multiErr, fmt.Errorf("no documents found in %s data stream", dataStream))
```

Comment:
Hmm .. What I thought is - In this case where we run system tests if there are no documents passed it means there is an error in the pipeline. We have the data for sure in the files but it didn't match the conditions of the pipeline.

I'll have another look at the logic from a few calls above the call stack to understand how else this can be handled. Thanks

Reply:
> Hmm .. What I thought is - In this case where we run system tests if there are no documents passed it means there is an error in the pipeline

My understanding is that if we can don't have any documents, we shouldn't call the validation of documents :)

Reply:
> > Hmm .. What I thought is - In this case where we run system tests if there are no documents passed it means there is an error in the pipeline
> 
> My understanding is that if we can don't have any documents, we shouldn't call the validation of documents :)

> > Hmm .. What I thought is - In this case where we run system tests if there are no documents passed it means there is an error in the pipeline
> 
> My understanding is that if we can don't have any documents, we shouldn't call the validation of documents :)

Thanks :). I think I see what you mean. I agree `validateFields` needn't be invoked. The runner invokes tests per each variant. For a case where there were no doc hits if an error is returned then it would not run tests for other variants. As you see here - https://github.com/elastic/elastic-package/blob/main/internal/testrunner/runners/system/runner.go#L176-L179. Do you think that's acceptable behaviour? 

I've modified the code.

Reply:
Yes, it's much shorter now :) If it fixes the panic, then it's 👍 . Thank you for the fix, Sai!